classes/webHookDispatcher/OrderDispatcher.php

```diff
@@ -191,4 +173,56 @@ private function dispatchPaymentStatus($eventType, $resource, $orderId)
 
         return true;
     }
+
+    /**
+     * @param string $eventType
+     * @param array $resource
+     * @param int $currentOrderStateId
+     *
+     * @return int
+     */
+    private function getNewState($eventType, $resource, $currentOrderStateId)
+    {
+        if (static::PS_CHECKOUT_PAYMENT_AUTH_VOIDED === $eventType) {
+            return (int) \Configuration::getGlobalValue('PS_OS_CANCELED');
+        }
+
+        if (static::PS_CHECKOUT_PAYMENT_COMPLETED === $eventType) {
+            return $this->getPaidStatusId($currentOrderStateId);
+        }
+
+        if (static::PS_CHECKOUT_PAYMENT_DENIED === $eventType) {
+            return (int) \Configuration::getGlobalValue('PS_OS_ERROR');
+        }
+
+        return $this->getPendingStatusId($resource);
+    }
+
+    /**
+     * @param int $currentOrderStateId Current OrderState identifier
+     *
+     * @return int OrderState paid identifier
+     */
+    private function getPaidStatusId($currentOrderStateId)
+    {
+        if ($currentOrderStateId === (int) \Configuration::getGlobalValue('PS_OS_OUTOFSTOCK_UNPAID')) {
+            return (int) \Configuration::getGlobalValue('PS_OS_OUTOFSTOCK_PAID');
+        }
+
+        return (int) \Configuration::getGlobalValue('PS_OS_PAYMENT');
+    }
```

Comment:
This method is too close to `getPaidStatusId` in classes/ValidateOrder.php and must be merged in the next version with it to avoid duplicated code.

Reply:
Will be a part of a new Service in 2.0.0 :)